components/selection/selection-mixin.js

```diff
@@ -57,7 +57,7 @@ export const SelectionMixin = superclass => class extends RtlMixin(CollectionMix
 		super();
 		this.selectionSingle = false;
 		this._selectAllPages = false;
-		this._selectionObservers = new Map();
+		this._selectionObservers = new Set();
```

Comment:
Changed this to a set because there was no need for a key and sets are faster.

Reply:
Actually, this inadvertently fixes another bug. where clicking `d2l-selection-select-all-pages` in `d2l-selection-controls` is fails to focus the `d2l-selection-select-all`.

Reply:
Ok, took way longer than I wanted it to (a lot has changed in almost two years - go figure) but here's an updated POC: https://github.com/BrightspaceUI/core/pull/3819

A lot of registries won't care what order their subscribers are in, so if there's even a slight perf cost I think it makes sense to keep it outside the subscriber controller code.  But just wanted to show what that would look like - it's currently an array of elements, there's no path info stored with it.